From 70f1f3c92363364544843c26c6ecf4555a89f0b4 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Tue, 16 May 2017 12:35:54 +0200
Subject: [PATCH] src: Avoid returning uninitialized data

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1353319
Upstream Status: libnftnl commit 7a150cacc1754

commit 7a150cacc1754f66525f0e87b14b35fbc2a6338e
Author: Phil Sutter <phil@nwl.cc>
Date:   Fri Aug 12 01:33:38 2016 +0200

    src: Avoid returning uninitialized data

    Although the 'err' pointer should be interesting for users only if the
    parser returned non-zero, having it point to uninitialized data is
    generally a bad thing.

    Signed-off-by: Phil Sutter <phil@nwl.cc>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
---
 src/chain.c | 2 +-
 src/rule.c  | 2 +-
 src/set.c   | 2 +-
 src/table.c | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/chain.c b/src/chain.c
index c956cec..ff87d1b 100644
--- a/src/chain.c
+++ b/src/chain.c
@@ -801,7 +801,7 @@ static int nftnl_chain_do_parse(struct nftnl_chain *c, enum nftnl_parse_type typ
 			      enum nftnl_parse_input input)
 {
 	int ret;
-	struct nftnl_parse_err perr;
+	struct nftnl_parse_err perr = {};
 
 	switch (type) {
 	case NFTNL_PARSE_XML:
diff --git a/src/rule.c b/src/rule.c
index b0fe30d..f202d03 100644
--- a/src/rule.c
+++ b/src/rule.c
@@ -689,7 +689,7 @@ static int nftnl_rule_do_parse(struct nftnl_rule *r, enum nftnl_parse_type type,
 			     enum nftnl_parse_input input)
 {
 	int ret;
-	struct nftnl_parse_err perr;
+	struct nftnl_parse_err perr = {};
 
 	switch (type) {
 	case NFTNL_PARSE_XML:
diff --git a/src/set.c b/src/set.c
index cc49891..7e44769 100644
--- a/src/set.c
+++ b/src/set.c
@@ -739,7 +739,7 @@ static int nftnl_set_do_parse(struct nftnl_set *s, enum nftnl_parse_type type,
 			    enum nftnl_parse_input input)
 {
 	int ret;
-	struct nftnl_parse_err perr;
+	struct nftnl_parse_err perr = {};
 
 	switch (type) {
 	case NFTNL_PARSE_XML:
diff --git a/src/table.c b/src/table.c
index b58ce73..6eb344e 100644
--- a/src/table.c
+++ b/src/table.c
@@ -359,7 +359,7 @@ static int nftnl_table_do_parse(struct nftnl_table *t, enum nftnl_parse_type typ
 			      enum nftnl_parse_input input)
 {
 	int ret;
-	struct nftnl_parse_err perr;
+	struct nftnl_parse_err perr = {};
 
 	switch (type) {
 	case NFTNL_PARSE_XML:
-- 
1.8.3.1

